Source/Core/Gen4/Generators/WildGenerator4.cpp

```diff
@@ -352,82 +353,62 @@ std::vector<WildState4> WildGenerator4::generateMethodK(u32 seed, const Encounte
             state.setLevel(encounterArea.calcLevel(state.getEncounterSlot(), go.nextUShort<true>()));
             occidentary += 1; // Compensate for the game's advances after the battle ends
             break;
+        case Encounter::BugCatchingContest:
+            loops = 4;
+            state.setEncounterSlot(EncounterSlot::kSlot(first, encounter));
+            if (!filter.compareEncounterSlot(state))
+            {
+                continue;
+            }
+
+            state.setLevel(0);
+            go.nextUShort<true>();
+            break;
         case Encounter::HeadButt: // TODO
-        case Encounter::BugCatchingContest: // TODO
         default:
             break;
         }
 
         u32 pid = 0;
-        switch (lead)
+        for (u8 loop = 0; loop < loops; loop++)
```

Comment:
I'm not sure if this is the best approach. In the successful cute charm case, we skip the loop as well for bug contest

Might need to refactor this section to leave the non bug contest case the same and better handle bug contest

Reply:
Rather than duplicating a ton of lines for contest vs non-contest is place a loop within each section that would actually loop. More lines than one loop but more lines that duplicating the whole switch statement.

Reply:
> I'm not sure if this is the best approach. In the successful cute charm case, we skip the loop as well for bug contest
> 
> Might need to refactor this section to leave the non bug contest case the same and better handle bug contest

How do we skip the loop in the cutecharm case? The if should avoid this

Reply:
> Rather than duplicating a ton of lines for contest vs non-contest is place a loop within each section that would actually loop. More lines than one loop but more lines that duplicating the whole switch statement.

I don't think you can have multiple loop section, cause you have to loop until you find an IV at 31, max 4 times. I think you can't separate the loop

Reply:
> I'm not sure if this is the best approach. In the successful cute charm case, we skip the loop as well for bug contest
> 
> Might need to refactor this section to leave the non bug contest case the same and better handle bug contest

Done

Reply:
What was the conclusion here?

Reply:
Refactored to remove duplication. Don't think extra functions are necessary. It may look more readable but the effect is still the same so I wouldn't say it's high priority.